### ISSUE TITLE
Psalm fails to report certain fatal error when it sees method_exists()

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1231]
Following from the discussion in #1230 

https://getpsalm.org/r/9473dc9153

```php
<?php
function f(object $o): void { 
  if (!method_exists($o, 'method')) {
    $o->method(); // Certain fatal error at runtime
  }
}
```
Expected: UndefinedMethod
Actual: no issues reported

**Comment by weirdan** [comment]
Perhaps it could also be `like-type<interface { function method(...$args); }>`. Assuming `like-type` (vimeo/psalm#1166) and something like anonymous interface. Visibility modifiers might be a problem though.

**Comment by muglug** [comment]
Not going to fix this - sort of beyond Psalm's understanding
